model/version.go

```diff
@@ -593,6 +597,90 @@ func GetVersionsWithOptions(projectName string, opts GetVersionsOptions) ([]Vers
 	return res, nil
 }
 
+// constructManifest will construct a manifest from the given project and version.
+func constructManifest(v *Version, projectRef *ProjectRef, moduleList ModuleList, settings *evergreen.Settings) (*manifest.Manifest, error) {
```

Comment:
Sorry to harp on the construction of this function too much but I think constructManifest and mergeWithBaseManifest can just be combined.
Instead of passing in just the auto-update modules to constructManifest, could we just pass in all of them? And then at 622 if we're looking at a patch and it's not auto-update, we use the base manifest (if I'm remembering the behavior correctly), otherwise we do the steps that exist here. 
Having the two helper functions that require very specific input (and rely on each other's specific output) seems like it'll make it bug prone in the future so it seems worth refactoring now if possible.